python_src/src/lamp_py/runtime_utils/infinite_wait.py

```diff
@@ -0,0 +1,26 @@
+import logging
+import time
+
+
+def infinite_wait(reason: str) -> None:
+    """
+    When running on ECS, propagating an exception up the call stack and killing
+    the processes will result in the process being restarted, to keep the task
+    count at one. This method should be called instead when we want to pause
+    the process for intervention before restarting.
+    """
+    # max amount of time to sleep between logging statements
+    max_sleep = 30 * 60
+
+    # initial sleep period, to be doubled between each log statement
+    nap_time = 60
+
+    while True:
+        # log the error
+        logging.error("Pausing for %s", reason)
```

Comment:
We should probably be running `check_for_sigterm` in this loop, to handle a graceful shutdown, right?